docs/components/MainSectionCard.js

```diff
@@ -63,20 +63,23 @@ const MainSectionCard = ({
   const shouldShowCode = showCode && cardSize !== 'sm' && type === 'info';
   const showTitleAndDescriptionAboveExample = cardSize === 'lg' && type === 'info';
 
-  const PreviewCard = ({ children: cardChildren }: PreviewCardProps): Node => (
-    <Box
-      alignItems="center"
-      borderStyle="sm"
-      color={shaded ? 'lightGray' : 'white'}
-      display="flex"
-      height={CARD_SIZE_NAME_TO_PIXEL[cardSize]}
-      justifyContent="center"
-      padding={8}
-      position="relative"
-      rounding={2}
-    >
-      {cardChildren}
-    </Box>
+  const PreviewCard = useCallback(
```

Comment:
@dangerismycat wouldn't it be even better to move `PreviewCard` out of `MainSectionCard`?

Reply:
I started with that initially, but came back to this to see how it felt. There is value in defining a function within the scope of another function to take advantage of the parent scope without needing to pass in everything, especially to avoid needing to repeat type definitions. Something like this really doesn't feel like another component, but just a quick helper. For anything that felt like a separate component, I extracted the function.

I'd like to enable this rule in Pinboard someday, so I wanted to explore this pattern. It feels ok to me — retaining the parent scope convenience while also making sure the function is stable.